pkg/ccl/importccl/import_pgdump.go

```diff
@@ -0,0 +1,503 @@
+						return err
+					}
+					// We have some sql. Parse and process it.
+					fnStmts, err := parser.Parse(fnSQL)
+					if err != nil {
+						return err
+					}
+					for _, fnStmt := range fnStmts {
+						switch ast := fnStmt.AST.(type) {
+						case *tree.AlterTable:
+							alterTableHandler := &postgresDDLHandler{}
+							err := bufferDDLPostgresStatement(ctx, evalCtx, ast, p, parentID, alterTableHandler)
+							if err != nil {
+								return err
+							}
+							handler.bufferedDDLStmts = append(handler.bufferedDDLStmts, alterTableHandler.bufferedDDLStmts[0])
+						default:
+							// We only support ALTER statements returned from a SQLFn.
+							return errors.Errorf("unsupported statement: %s", stmt)
+						}
+					}
+				default:
+					err := errors.Errorf("unsupported %T SELECT expr: %s", expr, expr)
+					return wrapErrorWithUnsupportedHint(err)
+				}
+			}
+		default:
+			err := errors.Errorf("unsupported %T SELECT %s", sel, sel)
+			return wrapErrorWithUnsupportedHint(err)
+		}
+	case *tree.DropTable:
+		names := stmt.Names
+
+		// If we find a table with the same name in the target DB we are importing
+		// into and same public schema, then we throw an error telling the user to
+		// drop the conflicting existing table to proceed.
+		// Otherwise, we silently ignore the drop statement and continue with the import.
+		for _, name := range names {
+			tableName := name.ToUnresolvedObjectName().String()
+			if err := p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+				err := catalogkv.CheckObjectCollision(
+					ctx,
+					txn,
+					p.ExecCfg().Codec,
+					parentID,
+					keys.PublicSchemaID,
+					tree.NewUnqualifiedTableName(tree.Name(tableName)),
+				)
+				if err != nil {
+					return errors.Wrapf(err, `drop table "%s" and then retry the import`, tableName)
+				}
+				return nil
+			}); err != nil {
+				return err
+			}
+		}
+	case *tree.BeginTransaction, *tree.CommitTransaction:
+	case *tree.Insert, *tree.CopyFrom, *tree.Delete, copyData:
+		// handled during the data ingestion pass.
+	case *tree.CreateExtension, *tree.CommentOnDatabase, *tree.CommentOnTable,
+		*tree.CommentOnIndex, *tree.CommentOnConstraint, *tree.CommentOnColumn, *tree.SetVar, *tree.Analyze,
+		*tree.CommentOnSchema:
+		// These are the statements that can be parsed by CRDB but are not
+		// supported, or are not required to be processed, during an IMPORT.
+		// - ignore txns.
+		// - ignore SETs and DMLs.
+		// - ANALYZE is syntactic sugar for CreateStatistics. It can be ignored
+		// because the auto stats stuff will pick up the changes and run if needed.
+		return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T statement: %s", stmt, stmt))
+	case *tree.CreateType:
+		return errors.New("IMPORT PGDUMP does not support user defined types; please" +
+			" remove all CREATE TYPE statements and their usages from the dump file")
+	case error:
+		if !errors.Is(stmt, errCopyDone) {
+			return stmt
+		}
+	default:
```

Comment:
I'm a little sad about that we're still doing this big switch that has to know about and handle (to a degree) every statement we could encounter, digging into the guts of some of them or ignoring a bunch of other ones like comments or whatever, so we're still going to be on the hook for at least some amount of import compatibilty stuff here vs just leaving it to SQL.

But, maybe i'm letting perfect be the enemy of good, and i mean, very good, since at least we're not _evaluating_ the DDLs anymore, and I don't immediately see a good alternative -- the switch and this default fail is good in that it makes sure we don't mistakenly miss rewriting a name in some other statment type.